From 5990e08f818ceca2534d082193085d42f2404c82 Mon Sep 17 00:00:00 2001
From: Richard Moe Gustavsen <richard.gustavsen@qt.io>
Date: Fri, 24 Oct 2025 15:22:02 +0200
Subject: [PATCH 212/239] SpinBox, macOS: make the stepper smaller

This is a 'cherry-pick' of
qtbase:b1a7a9d4f437c905df8368742b4189f1e8cea311

On macOS 26 this thing became rather large, not fitting into the
simple clipping rectangle of a spinbox (in a native app such a stepper is
taller than its controlled input, e.g. date/time edit). In Qt, with clipping
applied, this looks not so nice; but by always setting the cell's control size
to 'mini', we can have a properly looking spinbox, similar to what it was before.

Task-number: QTBUG-140134
Pick-to: 6.9 6.8 6.5
Change-Id: Ib157dc2f483a4fb10a4d99a68798367e0cd99a7a
Reviewed-by: Timur Pocheptsov <timur.pocheptsov@qt.io>
(cherry picked from commit 14ca47d0f10bc6657260d5f014896ded05fd7d9f)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quicknativestyle/qstyle/mac/qquickmacstyle_mac.mm | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/quicknativestyle/qstyle/mac/qquickmacstyle_mac.mm b/src/quicknativestyle/qstyle/mac/qquickmacstyle_mac.mm
index 6e5251aecc..b97c2e9844 100644
--- a/src/quicknativestyle/qstyle/mac/qquickmacstyle_mac.mm
+++ b/src/quicknativestyle/qstyle/mac/qquickmacstyle_mac.mm
@@ -4704,6 +4704,9 @@ void QMacStyle::drawComplexControl(ComplexControl cc, const QStyleOptionComplex
                 const auto cw = QMacStylePrivate::CocoaControl(QMacStylePrivate::Stepper, aquaSize);
                 NSStepperCell *cell = static_cast<NSStepperCell *>(d->cocoaCell(cw));
                 cell.enabled = (sb->state & State_Enabled);
+                const auto controlSize = cell.controlSize;
+                 if (qt_apple_runningWithLiquidGlass())
+                     cell.controlSize = NSControlSizeMini;
 
                 const CGRect newRect = [cell drawingRectForBounds:updown.toCGRect()];
 
@@ -4723,6 +4726,8 @@ void QMacStyle::drawComplexControl(ComplexControl cc, const QStyleOptionComplex
                     [cell stopTracking:pressPoint at:pressPoint inView:d->backingStoreNSView mouseIsUp:NO];
 
                 d->restoreNSGraphicsContext(cg);
+                if (qt_apple_runningWithLiquidGlass())
+                    cell.controlSize = controlSize;
             }
         }
         break;
-- 
2.51.2

